frontend/components/search/SearchForm.jsx

```diff
@@ -0,0 +1,158 @@
+import axios from "axios";
+
+import {
+  useState,
+  useContext,
+  createContext,
+  useCallback,
+  useEffect
+} from "react";
+import {useRouter} from "next/router";
+import {Container, Grid} from "@material-ui/core";
+import {makeStyles} from "@material-ui/core";
+
+import MultiSelect from "./MultiSelect";
+import CheckboxGroup from "./CheckboxGroup";
+import SearchButton from "./SearchButton";
+import YearSelect from "./YearSelect";
+import TextField from "./StyledTextField";
+import RadioGroup from "./RadioGroup";
+
+import PageTransitionContext from "../PageTransitionContext";
+
+export const SearchFormContext = createContext({});
+
+function useSearchFormState() {
+  // This hook is used to centralize the state of all search form inputs.
+
+  const router = useRouter();
+
+  // load the initial state from url query params
+  const [state, setState] = useState(router.query);
+
+  // event handler used by several inputs to set their state
+  const setStateFromEvent = useCallback(({target}) => setState(
+    (prevState) => ({...prevState, [target.name]: target.value})
+  ), []);
+
+  useEffect(() => {
+    // Remove any params we don't want sent to the next search page
+    // and update form state when browser history is navigated.
+    const {page, ...query} = router.query;
+    setState(query);
+  }, [router.query]);
+
+  // used to disable the entire form when page transition are occurring
+  const isLoading = useContext(PageTransitionContext);
+
+  return {state, setState, setStateFromEvent, disabled: isLoading};
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    paddingTop: "20px",
+    maxWidth: ({contained}) => contained ? undefined : theme.breakpoints.values.sm,
+    "& input": {
+      backgroundColor: "white",
+    },
+    "& .MuiTextField-root": {
+      backgroundColor: "white",
+    },
+    "& .MuiRadio-root, & .MuiCheckbox-root": {
+      marginBottom: "-9px",
+      marginTop: "-9px",
+    },
+    // to prevent hidden search button when navbar is visible
+    "&:last-child": {marginBottom: "50px"}
+  },
+}));
+
+export default function SearchForm({contained}) {
+  const classes = useStyles({contained});
+  const router = useRouter();
+  const formState = useSearchFormState();
+
+  // This value is used to change how the form is displayed
+  // in a sidebar vs when loaded by itself.
+  // See: https://material-ui.com/components/grid/#grid-with-breakpoints
+  const sm = contained ? 12 : 6;
```

Comment:
Why `contained`?  `contained` seems to imply that the component is being used inside a `Container`.  Is that the intent?

Reply:
Updated comments & variable name for clarity